File: task/src/main/java/com/goexp/galgame/data/piplline/handler/Starter.java
package com.goexp.galgame.data.piplline.handler;

import com.goexp.galgame.data.piplline.core.Message;
import com.goexp.galgame.data.piplline.core.MessageQueueProxy;

import java.util.concurrent.BlockingQueue;

public interface Starter {
    void process(MessageQueueProxy<Message> msgQueue);
}

File: task/src/main/scala/com/goexp/galgame/data/piplline/handler/Starter.scala
package com.goexp.galgame.data.piplline.handler

import com.goexp.galgame.data.piplline.core.Message
import com.goexp.galgame.data.piplline.core.MessageQueueProxy
import java.util.concurrent.BlockingQueue

trait Starter {
  def process(): Unit

  def setQueue(proxy: MessageQueueProxy[Message[_]]): Unit
}